docs/rules-from-web-poet.rst

```diff
@@ -187,21 +203,103 @@ Take note that since ``SCRAPY_POET_OVERRIDES`` is structured as
-method if implemented. This might be handy to be able to access settings, stats,
-request meta, etc.
+Item Returns
+============
+
+scrapy-poet also supports a convenient way of asking for items directly. This
+is made possible by the ``to_return`` parameter of :class:`web_poet.rules.ApplyRule`.
+The ``to_return`` parameter specifies which item a page object is capable of
+returning for a given URL.
+
+Let's check out an example:
+
+.. code-block:: python
+
+    import attrs
+    import scrapy
+    from web_poet import WebPage, handle_urls, field
+    from scrapy_poet import DummyResponse
+
+
+    @attrs.define
+    class Product:
+        name: str
+
+
+    @handle_urls("example.com")
+    @attrs.define
+    class ProductPage(WebPage[Product]):
+
+        @field
+        def name(self) -> str:
+            return self.css("h1.name ::text").get("")
+
+
+    class MySpider(scrapy.Spider):
+        name = "myspider"
+
+        def start_requests(self):
+            yield scrapy.Request(
+                "https://example.com/products/some-product", self.parse
+            )
+
+        # We can directly ask for the item here instead of the page object.
+        def parse(self, response: DummyResponse, item: Product):
+            return item
+
+From this example, we can see that:
+
+    * Spider callbacks can directly ask for items as dependencies.
+    * The ``Product`` item instance directly comes from ``ProductPage``.
+    * This is made possible by the ``ApplyRule("example.com", use=ProductPage,
+      to_return=Product)`` instance created from the ``@handle_urls`` decorator
+      on ``ProductPage``.
+
+.. note::
+
+    The slightly longer alternative way to do this is by declaring the page
+    object itself as the dependency and then calling its ``.to_item()`` method.
+    For example:
+
+    .. code-block:: python
+
+        @handle_urls("example.com")
+        @attrs.define
+        class ProductPage(WebPage[Product]):
+            product_image_page: ProductImagePage
+
+            @field
+            def name(self) -> str:
+                return self.css("h1.name ::text").get("")
+
+            @field
+            async def image(self) -> Image:
+                return await self.product_image_page.to_item()
+
+
+        class MySpider(scrapy.Spider):
+            name = "myspider"
+
+            def start_requests(self):
+                yield scrapy.Request(
+                    "https://example.com/products/some-product", self.parse
+                )
+
+            async def parse(self, response: DummyResponse, product_page: ProductPage):
+                return await product_page.to_item()
+
+For more information about all the expected behavior for the ``to_return``
+parameter in :class:`web_poet.rules.ApplyRule`, check out web-poet's tutorial
+regarding :ref:`rules-item-class-example`.
+
+
+Registry
+========
+
+As mentioned above, scrapy-poet uses the registry from web-poet called
+:class:`web_poet.rules.RulesRegistry`.
```

Comment:
It seems this might cause some confusion. In web-poet we have RulesRegistry class, and a default instance of this class (web_poet.default_registry). Users can create multiple registries, by creating other instances of RulesRegistry. I think in web-poet [docs](https://web-poet.readthedocs.io/en/stable/page-objects/rules.html) "registry" means "RulesRegistry instance", while here "registry" means "RulesRegistry class".

So, when I see SCRAPY_POET_REGISTRY option, I'd probably expect that it allows to use registries other than default_registry. But what it does is that it allows to load all rules from default_registry (or from SCRAPY_POET_RULES), and then create some other class from these rules, instead of RulesRegstry. 

To avoid the confusion, we should at least rename the option to SCRAPY_POET_REGISTRY_CLASS. But actually I'm not sure what's the use case for this option. I can see use cases for using registries other than default_registry (which is not supported directly - you need to do it through SCRAPY_POET_RULES), but why would users want to change the registry class?

Reply:
We may also allow to pass registry instances to SCRAPY_POET_RULES. It might make SCRAPY_POET_REGISTRY redundant (as you can pass an instance of a subclass if needed), and also simplify some of the use cases, with  `default_registry.get_rules()`.

Reply:
https://github.com/scrapinghub/scrapy-poet/pull/88#discussion_r1081051564